modules/tracking/src/trackerSRDCF.cpp

```diff
@@ -0,0 +1,647 @@
+                filter_index += 1;
+            }
+        }
+    }
+
+    return filter_matrix;
+}
+
+void SRDCF::split_spectrum(const cv::Mat fft_matrix, std::vector<float>& real, std::vector<float>& imag){
+    //exact number of elemts plus some extra
+    real.clear();
+    imag.clear();
+
+    real.reserve(fft_matrix.rows*fft_matrix.cols/2+5);
+    imag.reserve(fft_matrix.rows*fft_matrix.cols/2+5);
+
+    int nRows = fft_matrix.rows;
+    int nCols = fft_matrix.cols;
+
+    real.push_back(fft_matrix.at<float>(0,0)); //DC component
+    imag.push_back(0.0); //always zero imag for DC
+
+    for(int row = 1; row < nRows-1; row = row + 2){
+        real.push_back(fft_matrix.at<float>(row,0));
+        imag.push_back(fft_matrix.at<float>(row+1,0));
+    }
+    if(nRows % 2 == 0){
+        real.push_back(fft_matrix.at<float>(nRows-1,0));
+        imag.push_back(0.0);
+    }
+
+
+    for(int row = 0; row < nRows; row++){
+        for(int col=1; col < fft_matrix.cols-1; col = col + 2){
+            real.push_back(fft_matrix.at<float>(row,col));
+            imag.push_back(fft_matrix.at<float>(row,col+1));
+        }
+    }
+
+    if(nCols % 2 == 0){
+        real.push_back(fft_matrix.at<float>(0,nCols-1));
+        imag.push_back(0);
+
+        for(int row = 1; row < nRows-1; row = row + 2){
+            real.push_back(fft_matrix.at<float>(row,nCols-1));
+            imag.push_back(fft_matrix.at<float>(row+1,nCols-1));
+        }
+
+        if( nRows % 2 == 0){
+            real.push_back(fft_matrix.at<float>(nRows-1,nCols-1));
+            imag.push_back(0);
+        }
+    }
+}
+
+std::vector<cv::Mat> SRDCF::fft2(const cv::Mat featureData){
+    std::vector<cv::Mat> channels(featureData.channels());
+    std::vector<cv::Mat> channelsf(featureData.channels());
+    cv::split(featureData,channels);
+
+    for(size_t i=0; i < channels.size(); ++i){
+      cv::Mat windowed;
+      cv::multiply(channels[i],window,windowed);
+      cv::dft(windowed,channelsf[i],0);
+    }
+
+    return channelsf;
+}
+
+
+
+std::vector<cv::Mat> SRDCF::construct_sample_data(std::vector<cv::Mat> feature_matrices){
+
+    const int nFeat = feature_matrices.size();
+    std::vector<cv::Mat> outer_products(nFeat*nFeat);
+    //compute outer product of features
+    for(size_t feat_row = 0; feat_row < feature_matrices.size(); feat_row++){
+        for(size_t feat_col =0; feat_col < feature_matrices.size(); feat_col++){
+            cv::mulSpectrums(feature_matrices[feat_col],feature_matrices[feat_row],outer_products[feat_col*nFeat+feat_row],0,true);
+        }
+    }
+    return outer_products;
+}
+
+SparseMat SRDCF::make_sparse_data(const std::vector<cv::Mat>& data){
+    const int feature_dim = data.size();
+
+    std::vector<cv::Mat> outer_products = construct_sample_data(data);
+    std::vector<float> real,imag;
+
+    int sparse_mat_size = 2 * real.size() * feature_dim;
```

Comment:
real.size() is actually 0 in this case, since you haven't initialized the real vector. This leads to exception thrown when you try to declare a SparseMat of 0 size in line 481.